src/components/TableComponent/TableComponent.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useState } from 'react';
+import './TableComponent.scss';
+import { TableRow } from './index';
+import { SearchTerms } from '../../types/types';
+import SearchIcon from '../../assets/icons/Search.svg';
+
+interface TableProps {
+  data: Array<Record<string, any>>;
```

Comment:
I think it would be a good idea to create a specific type instead of using any. This way, we get autocomplete and reduce the chances of making mistakes.

We could have something like this:
```ts
type TableData = {
  content: string;
  type: 'text' | 'link' | 'address'
}

interface TableProps {
  data: Array<Record<string, TableData>>;
}
```